src/main/java/cuchaz/enigma/network/DedicatedEnigmaServer.java

```diff
@@ -0,0 +1,148 @@
+package cuchaz.enigma.network;
+
+import com.google.common.io.MoreFiles;
+import cuchaz.enigma.*;
+import cuchaz.enigma.throwables.MappingParseException;
+import cuchaz.enigma.translation.mapping.EntryRemapper;
+import cuchaz.enigma.translation.mapping.serde.MappingFormat;
+import cuchaz.enigma.utils.Utils;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+public class DedicatedEnigmaServer extends EnigmaServer {
+
+	private final EnigmaProfile profile;
+	private final MappingFormat mappingFormat;
+	private final Path mappingsFile;
+	private final PrintWriter log;
+	private BlockingQueue<Runnable> tasks = new LinkedBlockingDeque<>();
```

Comment:
Can make this final

Reply:
Not strictly necessary, but I can I guess. I tend to set instance fields to final so I don't forget to set them in the constructor.